### DISCUSSION TITLE
Is there a configuration for Caddy?

### DISCUSSION THREAD

**Discussion by simon-pipe** [datenstrom/community#1004] [Ask a question] [answered]
Hello, I'd like to use Caddy as a web server for Yellow. I run Caddy and PHP8.2 and with phpinfo() in the index.php file I get an output as well. What I can't manage is that Yellow runs. Does anyone have it running and could show me their Caddyfile? Thank you very much and Happy New Year!

**Comment by GiovanniSalmeri** [comment]
Hello. I guess that the only problem is translating .htaccess into the Caddy syntax. I have found very similar questions in the Caddy forum:

https://caddy.community/t/help-converting-htaccess-over-to-caddy-redir/460
https://caddy.community/t/htaccess-to-caddy-v2/9270

There is also a PHP script which automatically translates, but it seems to me overkill for the very simple .htaccess file of Yellow:

https://github.com/buckybytes/htaccess-to-caddyfile-converter

Perhaps you could ask for help in the Caddy forum and then post the solution here? It coud be useful if there were instructions for using Yellow with various web servers (some time ago I successfully used Yellow with [lighttpd](https://www.lighttpd.net/)).

**Reply by gstrauss** [reply]
lighttpd.conf equivalent for the yellow `.htaccess` would look something like this (untested):
```
index-file.names = ("index.html")
server.modules += ("mod_access", "mod_rewrite")
if $HTTP["url"] =~ "^/(?:content|system)/" {
    url.access-deny = ("")
}
else {
    url.rewrite-if-not-file = ("" => "/yellow.php")
}
```
plus the `fastcgi.server` configuration to run .php.  (https://wiki.lighttpd.net/mod_fastcgi)

**Comment by simon-pipe** [comment, marked as answer]
I am a try and error programmer. :-)
This way I found out the following Caddyfile that runs on me:

```
:80 {
   php_fastcgi php:9000
	
   try_files {path} /index.html /yellow.php
    
   @notfound {
      not file
      not path_regexp blockDirs
   }
   rewrite @notfound /yellow.php

   @blocked {
      path /content/* /system/*
   }
   rewrite @blocked /home/error
    
   file_server
   root * /usr/share/caddy/yellow
}
```

If someone has improved, I am grateful for it.

**Comment by annaesvensson** [comment]
Usually your web server has to do three things:

1. Forward requests for non existing files/directories to the `yellow.php`
2. Block direct access to the `content` directory with an error page
3. Block direct access to the `system` directory with an error page

This is required for using Datenstrom Yellow as a content management system, for example when you want to edit your website in the web browser. Here's some technical background for website administrators. The first item on the list is what makes everything work together. This is where the [core](https://github.com/annaesvensson/yellow-core) and other extensions are loaded and then handle the incoming HTTP requests. The second item on the list makes sure that nobody sees your content files in the raw Markdown format. The third item on the list makes sure that nobody sees your system files and personal data. As you can see there is no special rule for the media directory, that's because media files such as images are handled as regular files. Some web servers require additional settings, for example for the FastCGI configuration, but this is very much dependent on the web server and the operating system. Everyone does it a little different. There are example configurations for the Apache web server and the Nginx web server. I think it shouldn't be too difficult to make it work with any other web server. If you're working on your own computer, you can also use the [built-in web server](https://github.com/annaesvensson/yellow-serve).

**Reply by simon-pipe** [reply]
Thank you very much for your detailed answer. I think I have implemented all the points you mentioned with the Caddyfile listed above. And with me, Yellow has been running very well with Caddy so far.
